src/node.cc

```diff
@@ -4978,6 +4990,324 @@ int Start(int argc, char** argv) {
+  cmd_args.argv = uv_setup_args(cmd_args.argc, cmd_args.argv);
+
+  // This needs to run *before* V8::Initialize().  The const_cast is not
+  // optional, in case you're wondering.
+  // Init() puts the v8 specific cmd args in exec_argc and exec_argv, but as we
+  // don't support these, they are not used.
+  int exec_argc = 0;
+  const char** exec_argv = nullptr;
+  Init(&cmd_args.argc, const_cast<const char**>(cmd_args.argv), &exec_argc, &exec_argv);
+
+  initialize::configureOpenSsl();
+
+  initialize::initV8();
+
+  //////////
+  // Start 2
+  //////////
+
+  initialize::createIsolate();
+
+  initialize::createInitialEnvironment();
+
+  //////////
+  // Start environment
+  //////////
+  _StartEnv(cmd_args.argc, (const char* const*)cmd_args.argv);
+}
+
+void _StartEnv(int argc,
+               const char* const* argv) {
+    std::cout << "Starting environment" << std::endl;
+    /*std::cout << "argv" << std::endl;
+    for (int i = 0; i < argc; i++) {
+      std::cout << argv[i] << std::endl;
+    }
+    std::cout << "exec_argv" << std::endl;
+    for (int i = 0; i < exec_argc; i++) {
+      std::cout << exec_argv[i] << std::endl;
+    }*/
+
+    int v8_argc = 0;
+    const char* const* v8_argv = nullptr;
+    env->Start(argc, argv, v8_argc, v8_argv, v8_is_profiling);
+
+    const char* path = argc > 1 ? argv[1] : nullptr;
+    StartInspector(env, path, debug_options);
+
+    if (debug_options.inspector_enabled() && !v8_platform.InspectorStarted(env)) {
+      return; // TODO: Handle error
+      //return 12;  // Signal internal error.
+    }
+
+    env->set_abort_on_uncaught_exception(abort_on_uncaught_exception);
+
+    if (no_force_async_hooks_checks) {
+      env->async_hooks()->no_force_checks();
+    }
+
+    {
+      callback_scope = new Environment::AsyncCallbackScope(env);
+      env->async_hooks()->push_async_ids(1, 0);
+      LoadEnvironment(env);
+      env->async_hooks()->pop_async_id(1);
+    }
+
+    env->set_trace_sync_io(trace_sync_io);
+}
+
+v8::Local<v8::Value> Run(const std::string& path) {
+  // Read entire file into string. There is most certainly a better way ;)
+  // https://stackoverflow.com/a/2602258/2560557
+  std::ifstream t(path);
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+
+  return Evaluate(buffer.str().c_str());
+}
+
+v8::Local<v8::Value> Evaluate(const std::string& java_script_code) {
+  EscapableHandleScope scope(env->isolate());
+  TryCatch try_catch(env->isolate());
+
+  // try_catch must be nonverbose to disable FatalException() handler,
+  // we will handle exceptions ourself.
+  try_catch.SetVerbose(false);
+
+  //ScriptOrigin origin(filename);
+  MaybeLocal<v8::Script> script = v8::Script::Compile(env->context(), v8::String::NewFromUtf8(isolate, java_script_code.c_str())/*, origin*/);
+  if (script.IsEmpty()) {
+    ReportException(env, try_catch);
+    exit(3);
+  }
+
+  Local<Value> result = script.ToLocalChecked()->Run();
```

Comment:
MaybeLocal::ToLocalChecked() [crashes the process](https://v8.paulfryzel.com/docs/master/classv8_1_1_maybe_local.html#a9b2c9d50fca5897e3a03fd4c25d12415). Isn't this a bit extreme? I think a ```IsEmpty()``` check + error handling might be cleaner here. I suppose this should be fine for now but we need to remember it for issue #17, hence the comment.

Reply:
You are correct - I didn't use it conciously, I simply copied a bunch of code from https://github.com/hpicgs/node/blob/7a59273174a67ef0c3e20d234417b8a460508337/src/node.cc#L1928-L1953.